cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/ckan/CKANBackendImpl.java

```diff
@@ -439,6 +440,157 @@ private boolean existsView(String resId) throws Exception {
         } // try catch
     } // existsView
     
+    private JSONObject getRecords(String resId, String filters) throws Exception {
+        try {
+            // create the CKAN request JSON
+            String jsonString = "{\"id\": \"" + resId + "\"";
+        
+            if (filters == null || filters.isEmpty()) {
+                jsonString += "}";
+            } else {
+                jsonString += ",\"filters\":\"" + filters + "\"}";
+            } // if else
+
+            // create the CKAN request URL
+            String urlPath = "/api/3/action/datastore_search";
+
+            // do the CKAN request
+            JsonResponse res = doCKANRequest("POST", urlPath, jsonString);
+
+            // check the status
+            if (res.getStatusCode() == 200) {
+                LOGGER.debug("Successful search (resourceId=" + resId + ")");
+                return res.getJsonObject();
+            } else {
+                throw new CygnusPersistenceError("Could not search for the records (resId=" + resId
+                        + ", statusCode=" + res.getStatusCode() + ")");
+            } // if else if else
+        } catch (Exception e) {
+            if (e instanceof CygnusRuntimeError
+                    || e instanceof CygnusPersistenceError
+                    || e instanceof CygnusBadConfiguration) {
+                throw e;
+            } else {
+                throw new CygnusRuntimeError(e.getMessage());
+            } // if else
+        } // try catch
+    } // getRecords
+    
+    private void deleteRecords(String resId, String filters) throws Exception {
+        try {
+            // create the CKAN request JSON
+            String jsonString = "{\"id\": \"" + resId + "\",\"force\":\"true\"";
+        
+            if (filters == null || filters.isEmpty()) {
+                jsonString += "}";
+            } else {
+                jsonString += ",\"filters\":" + filters + "}";
+            } // if else
+
+            // create the CKAN request URL
+            String urlPath = "/api/3/action/datastore_delete";
+
+            // do the CKAN request
+            JsonResponse res = doCKANRequest("POST", urlPath, jsonString);
+
+            // check the status
+            if (res.getStatusCode() == 200) {
+                LOGGER.debug("Successful deletion (resourceId=" + resId + ")");
+            } else {
+                throw new CygnusPersistenceError("Could not delete the records (resId=" + resId
+                        + ", statusCode=" + res.getStatusCode() + ")");
+            } // if else if else
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1351/commits/733214a6c45b3629c8054f3043d452340cf06c84